### DISCUSSION TITLE
Bots using functions like forward() are getting stuck.

### DISCUSSION THREAD

**Discussion by TheFlameFish** [robocode-dev/tank-royale#87] [Q&A] [answered]
Hello. I've just started and I'm noticing that the bot I programmed via the MyFirstBot tutorial is frequently getting stuck for no apparent reason. While using the forward() or back() functions it begins sort of vibrating and is stuck like that until another tank rams it. I've also noticed this issue in some of the other sample bots. Any help is appreciated.

```Java
import dev.robocode.tankroyale.botapi.*;

public class MyFirstBot extends Bot {

    // The main method starts our bot
    public static void main(String[] args) {
        new MyFirstBot().start();
    }

    // Constructor, which loads the bot config file
    MyFirstBot() {
        super(BotInfo.fromFile("MyFirstBot.json"));
    }

    // Called when a new round is started -> initialize and do some movement
    @Override
    public void run() {
        // Repeat while the bot is running
        while (isRunning()) {
            forward(100);
            turnGunRight(360);
            
            back(100);
            turnGunRight(360);
        }
    }
}
```

**Comment by flemming-n-larsen** [comment, marked as answer]
@TheFlameFish:
Thank you for reporting this. 😊👌
I believe this bug was introduced with version 0.23.0, and I made a fix for it with the next version 0.23.1.

A temporary work-around would be to stick to use [version 0.22.2](https://github.com/robocode-dev/tank-royale/releases/tag/v0.22.2) until the next version is out, which does not have this bug.

**Reply by TheFlameFish** [reply]
Ah, Okay. Thank you.

**Reply by flemming-n-larsen** [reply]
@TheFlameFish 
[Version 0.23.1](https://github.com/robocode-dev/tank-royale/releases/tag/v0.23.1) with a fix has just been released. Please create a bug report, if the bug did not disappear. 😊
